src/attr/_funcs.py

```diff
@@ -72,19 +72,23 @@ def asdict(
                 )
             elif isinstance(v, (tuple, list, set, frozenset)):
                 cf = v.__class__ if retain_collection_types is True else list
-                rv[a.name] = cf(
-                    [
-                        _asdict_anything(
-                            i,
-                            is_key=False,
-                            filter=filter,
-                            dict_factory=dict_factory,
-                            retain_collection_types=retain_collection_types,
-                            value_serializer=value_serializer,
-                        )
-                        for i in v
-                    ]
-                )
+                items = [
+                    _asdict_anything(
+                        i,
+                        is_key=False,
+                        filter=filter,
+                        dict_factory=dict_factory,
+                        retain_collection_types=retain_collection_types,
+                        value_serializer=value_serializer,
+                    )
+                    for i in v
+                ]
+                try:
+                    rv[a.name] = cf(items)
+                except TypeError:
+                    # Workaround for TypeError: cf.__new__() missing 1 required
+                    # positional argument (which appears, for a namedturle)
+                    rv[a.name] = cf(*items)
```

Comment:
Thanks for looking into this!

I think I'm not comfortable with such a broad treatment of the rather common TypeError. The risk of masking other problems is too high.

I would prefer if you'd add something to the effect of `if not isinstance(cf, TypedDict): raise` before it

Reply:
Are you sure that it is should be `TypedDict`? The problem concerns `typing.NamedTuple` (and `collection.namedtuple`).
And unfortunately, it can not be determined using `isinstance`.
Except like this:
```python
if isinstance(v, tuple) and hasattr(v, '_fields'/'_make'/'_replace'/'_asdict')
```

Reply:
hmm, given tuple works OOB, shouldn't a TypeError + `isinstance(cf, tuple)` work then?

Reply:
> hmm, given tuple works OOB, shouldn't a TypeError + `isinstance(cf, tuple)` work then?

I thought so too, but it doesn't work:
```bash
FAILED tests/test_funcs.py::TestAsDict::test_recurse_property - AssertionError: assert False
FAILED tests/test_funcs.py::TestAsDict::test_lists_tuples - TypeError: list expected at most 1 argument, got 3
FAILED tests/test_funcs.py::TestAsDict::test_lists_tuples_retain_type - TypeError: tuple expected at most 1 argument, got 3
FAILED tests/test_funcs.py::TestAsTuple::test_recurse_retain - KeyError: 0
FAILED tests/test_funcs.py::TestAsTuple::test_lists_tuples - TypeError: list expected at most 1 argument, got 3
FAILED tests/test_funcs.py::TestAsTuple::test_lists_tuples_retain_type - TypeError: tuple expected at most 1 argument, got 3
```

Reply:
I made your wish come true with `issabclass`
```python
try:
    ...
except TypeError:
    if not issubclass(cf, tuple):
        raise 
    # workaroung
```
Thanks for the tip!

Reply:
Awesome!

Can you pls get rid of the `no cover`s? 😇 Unless you've got a good reason? Shouldn't be too hard to emulate?

Reply:
I'll try :smiley:

Reply:
We always believed in you!